src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java

```diff
@@ -85,81 +85,80 @@ protected void internalTransform(String phaseName,
 		hierarchy = scene.getActiveHierarchy();
 
 		// iterate through application classes, rename methods with junk
-		for (SootClass c : scene.getApplicationClasses()) {
-			Vector<String> fields = new Vector<String>();
-			Iterator<SootField> fIt = c.getFields().iterator();
-			while (fIt.hasNext()) {
-				fields.add(fIt.next().getName());
-			}
+		for (SootClass sc : scene.getApplicationClasses()) {
+			List<String> fieldNames = sc.getFields().stream().map(SootField::getName).collect(toList());
+			List<SootMethod> methods = new ArrayList<>(sc.getMethods());
 
-            final List<SootMethod> methods = new ArrayList<>(c.getMethods());
-            for (SootMethod m : methods) {
-				String subSig = m.getSubSignature();
+			for (SootMethod method : methods) {
+				String subSig = method.getSubSignature();
 
-				if (!allowsRename(c, m))
+				if (!allowsRename(sc, method)) {
 					continue;
+				}
 
 				boolean rename = true;
-				for (SootClass _c : hierarchy.getSuperclassesOfIncluding(c
-						.getSuperclass())) {
-					if (_c.declaresMethod(subSig)
-							&& hierarchy.isVisible(c, _c.getMethod(subSig))
-							&& _c.isLibraryClass()) {
-						if (output)
-							out.println("\t" + _c.getName() + "'s method "
+				//use getSuperclassesOfIncluding instead of getSuperclassesOf to avoid problems when sc is an interface
+				for (SootClass c : hierarchy.getSuperclassesOfIncluding(sc.getSuperclass())) {
+					if (c.declaresMethod(subSig)
+							&& hierarchy.isVisible(sc, c.getMethod(subSig))
+							&& c.isLibraryClass()) {
+						if (output) {
+							out.println("\t" + c.getName() + "'s method "
 									+ subSig + " is overridden in "
-									+ c.getName());
+									+ sc.getName());
+						}
 						rename = false;
 						break;
 					}
 				}
 
 				if (rename) {
-					// TODO: This is flawed since it all methods of a similar
```

Comment:
Where you able to fix t this? Where?

Reply:
I will provide commit for this in a while (after I check changes on my project and ensure that all changes work fine), for now I have returned this TODO in f31aba60e76f438973c61abcf85d78b2e33d5392